File: androidApp/src/main/kotlin/app/futured/kmptemplate/android/tools/binding/PlatformBindingsImpl.kt
package app.futured.kmptemplate.android.tools.binding

import android.content.Context
import app.futured.kmptemplate.platform.binding.Platform
import app.futured.kmptemplate.platform.binding.PlatformBindings
import app.futured.kmptemplate.platform.binding.PlatformFirebaseCrashlytics

class PlatformBindingsImpl(private val applicationContext: Context) : PlatformBindings {
    override fun platform(): Platform = PlatformImpl(applicationContext)
    override fun firebaseCrashlytics(): PlatformFirebaseCrashlytics = PlatformFirebaseCrashlyticsImpl()
}

File: iosApp/iosApp/Platform/PlatformBindingsImpl.swift
import Foundation
import KMP

final class PlatformBindingsImpl: PlatformBindings {

    func platform() -> any Platform {
        PlatformImpl()
    }

    func firebaseCrashlytics() -> PlatformFirebaseCrashlytics {
        PlatformFirebaseCrashlyticsImpl()
    }
}
